framework_tvb/tvb/interfaces/rest/client/user/user_api.py

```diff
@@ -28,23 +28,36 @@
 #
 #
 
-import requests
-
 from tvb.interfaces.rest.client.client_decorators import handle_response
 from tvb.interfaces.rest.client.main_api import MainApi
-from tvb.interfaces.rest.commons import RestNamespace, RestLink, LinkPlaceholder
-from tvb.interfaces.rest.commons.dtos import UserDto, ProjectDto
+from tvb.interfaces.rest.commons.strings import RestLink
+from tvb.interfaces.rest.commons.dtos import ProjectDto
 
 
 class UserApi(MainApi):
     @handle_response
-    def get_users(self):
-        response = requests.get(self.build_request_url(RestNamespace.USERS.value))
-        return response, UserDto
+    def login(self, username, password):
+        response = self.secured_request().post(self.build_request_url(RestLink.LOGIN.compute_url(True)), json={
+            "username": username,
+            "password": password
+        })
+        return response
+
+    @handle_response
+    def refresh(self, refresh_token):
```

Comment:
Could you document on this method when and how should be used?
is BrainX3 expected to call it, or we do it automatically?

Reply:
I'm refactoring this part so refreshing token will be done automatically before a request if it is expired

Reply:
I see this had been done, and it is happening as long as one of out TVB Rest API methods is called. I think this is the only appropriate way to handle this.